### ISSUE TITLE
Using AceJump while text is selected in Editor

### ISSUE THREAD

**Issue by suamikim** [johnlindquist/AceJump#9]
Following procedure:
- Select some text in the editor
- Use AceJump to jump somewhere else
- The cursor jumps to the specified position, the previously selected text stays selected (so far everything is fine!)
- If i start to type now or try to move the cursor with the arrow-keys the cursor jumps back to the place where the selected text is. IMHO the selected text should now be deselected and the cursor should stay at it's new position.

Usually this wouldn't be a problem but I work a lot like the following:
- Select some text, copy it with Ctrl+C
- Use AceJump to go to some position where i would like to insert the text
- Press Ctrl+V to insert it. And exactly this is not working because of the above mentioned behaviour (the cursor jumps back to the selection and won't insert the text at the new position)

I just started to use AceJump and really love it! Therefore I'd really appreciate if you could fix this because it'd makes some workflows so much easier!

My config: Win7, 64 Bit, WebStorm 5.0.4

Please let me know if you need any more information.

Thanks, mike

**Comment by jmdodge** [comment]
+2

**Comment by jmdodge** [comment]
Sorry for this but... pleeeeeeease with a cherry on top can you spare a moment to fix this?

**Comment by johnlindquist** [comment]
I've actually switched to promoting and helping with the development of
https://plugins.jetbrains.com/plugin/7163?pr=idea . It has many more
"AceJump" features than my plugin and is written much better. You'll have
to remap some of it's default keyboard shortcuts, but it should be an easy
drop-in replacement for this plugin.

**Comment by jmdodge** [comment]
Thanks for responding John.
I have tried the emacsIDEA plugin before and took the time to try it again.
I have a few issues with it - the documentation does not seem to be accurate, it's unclear what needs to be done to get all of it's features to work, frankly there are just more features than I need and it seems over complicated, I don't like the way the second target character is hidden if there are too many selection options (your plugin shows two-letter combinations - I've requested this before on the other project's github and got denied), and I find the way you work with line starts/endings to be cumbersome and unintuitive compared to this one ('space' key shows ALL line starts AND endings on the screen at once, it's a giant mess...and then you still have to reveal a second target key to hunt down when you go there).

In short, it's just unwieldy enough for me that I just drift back to arrow keys and mouse because it's less fuss most of the time.  Jumping just saves you a little bit of time, so if its not just right, it's easier to just ditch it.

Your plugin is lightweight and, save for this one little quirk, absolutely perfect for me.
I can't imagine this is a tough fix - maybe you could point me in the right direction if you'll still accept pull requests? I know squat about plugin development, but it looks like the text just needs to be deselected before the jump.  Maybe you'd consider a small payment to fix?

**Comment by johnlindquist** [comment]
I need to set up the plugin dev environment again. I kind of gave up for a
bit after running into some cross platform focus issues on nix/mac/win
which I really didn't want to deal with on a small personal side project.

I'll try to fire it up again this week to see what I can find.

Which platform are you on? Win/Mac/Nix?

**Comment by jmdodge** [comment]
I'm on Win 7, but I pulled up an OS X machine and saw the same behavior. Again, looks like the text just stays selected after the caret moves - deselecting the text after a jump should solve it.

Thanks for your willingness to look in to this, you're a credit to humanity! Beers on me.
